### ISSUE TITLE
Option to merge geoms for AOI splitting during project creation 

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/fmtm#1282]
**Is your feature request related to a problem? Please describe.**

- Sometimes the AOI geojson is a feature collection or multipolygon.
- Splitting will fail if a multi geom is used.

**Describe the solution you'd like**

- Provide the option to convex hull the geoms.
- Probably via a checkbox the user ticks (as they may wish to use the multigeom as the actual task boundaries instead).

**Describe alternatives you've considered**

- Even better would be an actual merging of geom outlines, but this is complex if the boundaries do not perfectly align.
- The convex hull should possibly be in fmtm-splitter, the same as the one in osm-rawdata.

**Additional context**

Requirement from an actual mapping campaign in Liberia.

**Comment by spwoodcock** [comment]
Related to #1263

We should probably have a configuration section in place on the frontend if the user selects the splitting algorithm.

**Comment by Sujanadh** [comment]
I think I did this long ago using an approach to splitting each individual polygon from a multipolygon by passing a polygon at a time to the algorithm, which splits it and appending the results of every polygon.

**Comment by spwoodcock** [comment]
Yes I think the code is there to loop through geoms and return a list of FeatureCollections 👍

The issue is that often by processing separately the geom does not have enough polyline data within it to split correctly.

The best solution is improving the splitting algorithm so it doesn't need polylines. If none are present then the features are simply divided up using the 'num buildings' param.

We should probably also provide a configuration to merge the geoms before splitting, if the user wishes.

**Comment by Sujanadh** [comment]
For now, if user uploads two polygons where they are touched or overlapping as an AOI, and still chooses to split task we combine those polygons making a single polygon and allow splitting 
Again, if those polygons are disjoint we dissolve those polygons and create new boundary as an aoi and allow splitting if they choose task split.
Input Polygons
![image](https://github.com/hotosm/fmtm/assets/109404840/cdba0165-379e-471c-8c1f-ee5fb4b5c071)

Output 
![image](https://github.com/hotosm/fmtm/assets/109404840/e94c1e9f-1195-4abc-b985-a43beec6fde6)

task-split
![image](https://github.com/hotosm/fmtm/assets/109404840/739e61d3-a6fd-4370-aa57-3cbfe429338f)

**Comment by Sujanadh** [comment]
@spwoodcock

**Comment by spwoodcock** [comment]
Does this already exist? Nice if so, we can close the issue!

**Comment by spwoodcock** [comment]
Yeah I think this is all good 👍 

![image](https://github.com/hotosm/fmtm/assets/78538841/4b5b60aa-3e83-4b15-bc77-3c5b569bb29c)

I uploaded a multipolygon AOI and successfully generated an extract.

**Comment by Sujanadh** [comment]
yes, it does allow splitting but there is still an issue while creating the project with this type of project, since we don't accept multipolygon as an project AOI.

**Comment by Sujanadh** [comment]
There is slight issue while merging polygons; we perform `unary_union` if the polygon is mulitpolygon , but if the polygons are disjoint then it will again results into MultiPolygon type, so in order to handle that we need to dissolve those polygons and create new boundary with a single polygon.

**Comment by Sujanadh** [comment]
![image](https://github.com/hotosm/fmtm/assets/109404840/773abb65-aa1c-4636-8b0e-f26daa000919)
This what it looks like while splitting task using disjoint polygon.
if we want to dissolve the polygons and split it , it will look like this:
![image](https://github.com/hotosm/fmtm/assets/109404840/def65f0f-ce88-471c-baba-da564b88b410)
so if we want to handle splitting like the first one , we should allow db to save multipolygon aoi also.
if not we need to apply convex hull to dissolve the polygons and create a single Polygon like the second one.

**Comment by manjitapandey** [comment]
I think first type of merging covers most of the use cases. We can work on second type of cases if requested for any field mapping activity. But for now, i can't imagine scenarios where they upload disjoint polygons and want tasks to get merged.

**Comment by spwoodcock** [comment]
Its not common, granted, but it could he nicebatse point that the user uploads their custom data extract and the AOI is automatically generated by enveloping (convex hull / merging) the geoms

**Comment by spwoodcock** [comment]
Can be addressed in hotosm/geojson-aoi-parser
